src/sentry/tasks/weekly_reports.py

```diff
@@ -0,0 +1,808 @@
+        key_transactions = query_result["data"]
+        ctx.projects[project.id].key_transactions_this_week = [
+            (i["transaction_name"], i["count"], i["p95"]) for i in key_transactions
+        ]
+
+        # Query the p95 for those transactions last week
+        query = Query(
+            match=Entity("transactions"),
+            select=[
+                Column("transaction_name"),
+                Function("quantile(0.95)", [Column("duration")], "p95"),
+                Function("count", [], "count"),
+            ],
+            where=[
+                Condition(Column("finish_ts"), Op.GTE, ctx.start - timedelta(days=7)),
+                Condition(Column("finish_ts"), Op.LT, ctx.end - timedelta(days=7)),
+                Condition(Column("project_id"), Op.EQ, project.id),
+                Condition(
+                    Column("transaction_name"),
+                    Op.IN,
+                    [i["transaction_name"] for i in key_transactions],
+                ),
+            ],
+            groupby=[Column("transaction_name")],
+        )
+        request = Request(dataset=Dataset.Transactions.value, app_id="reports", query=query)
+        query_result = raw_snql_query(request, referrer="weekly_reports.key_transactions.last_week")
+
+        # Join this week with last week
+        last_week_data = {
+            i["transaction_name"]: (i["count"], i["p95"]) for i in query_result["data"]
+        }
+
+        ctx.projects[project.id].key_transactions = [
+            (i["transaction_name"], i["count"], i["p95"])
+            + last_week_data.get(i["transaction_name"], (0, 0))
+            for i in key_transactions
+        ]
+
+
+def project_key_performance_issues(ctx, project):
+    if not project.first_event:
+        return
+    if not features.has("organizations:performance-issues", ctx.organization):
+        return
+
+    with sentry_sdk.start_span(op="weekly_reports.project_key_performance_issues"):
+        # Pick the 50 top frequent performance issues last seen within a month with the highest event count from all time.
+        groups = Group.objects.filter(
+            project_id=project.id,
+            status=GroupStatus.UNRESOLVED,
+            last_seen__gte=ctx.end - timedelta(days=30),
+            # performance issue range
+            type__gte=1000,
+            type__lt=2000,
+        ).order_by("-times_seen")[:50]
+        groups = list(groups)
+        group_id_to_group = {group.id: group for group in groups}
+
+        if len(group_id_to_group) == 0:
+            return
+
+        # Fine grained query for 3 most frequent events happend during last week
+        query = Query(
+            match=Entity("transactions"),
+            select=[
+                Column("group_ids"),
+                Function("count", []),
+            ],
+            where=[
+                Condition(Column("finish_ts"), Op.GTE, ctx.start),
```

Comment:
What is `finish_ts`?

Reply:
the time that the transaction was finished.
Snuba requires that we provide boundary conditions on this column.